### ISSUE TITLE
ZZ: need long long conv, maybe

### ISSUE THREAD

**Issue by Artoria2e5** [libntl/ntl#28]
On Apple's macOS, both `long` and `long long` are 64 bits. The system chooses to define `uint64_t` as `unsigned long long`, which isn't covered by NTL's `conv` routines, leading to some pretty unpleasant errors as I detailed in https://github.com/vigna/CPRNG/issues/1.

Because NTL covers `unsigned long` which is the same size on that particular system, I ended up getting away with supplying an explicit `unsigned long` type argument to the template. But that's hacky. Maybe something can be done to officially add `long long` to ZZ.h?

**Comment by victorshoup** [comment]
I agree, it is something that should be addressed. On Mar 2, 2024, at 3:30 AM, Mingye Wang ***@***.***> wrote:﻿
On Apple's macOS, both long and long long are 64 bits. The system chooses to define uint64_t as unsigned long long, which isn't covered by NTL's conv routines, leading to some pretty unpleasant errors as I detailed in vigna/CPRNG#1.
Because NTL covers unsigned long, on that particular system I ended up getting away with supplying an explicit unsigned long type argument to the template. But that's hacky. Maybe something can be done to officially add long long to ZZ.h?
